src/assembler/encoder.ts

```diff
@@ -71,8 +70,8 @@ export function encode(code: ICodeFile): IELF {
       try {
         writeInstruction(writer, instruction, pool)
       } catch (e: any) {
-        if (e instanceof VirtualBoardError) {
-          throw new CompileError(instruction.line, e.message, e.type)
+        if (e instanceof Error) {
+          throw new AssemblerError(e.message, instruction.line)
```

Comment:
why don't you catch just InstructionError where the user can 
or make an alert or something similar to let the user know that the checked out verison probably has a bug.

Reply:
The problem with that is that for example if an instruction was not found in the instruction set, a normal error is thrown. If I would just catch InstructionError here, error would just be catched and everything would continue on. But what actually should happen is that the user gets the information that the instruction name is not known.

If this is what you meant.

Reply:
kind of I would make an else if for error and first just InstructionError. So InstructionError will be shown in red box and errors with MessageBox or Alert

Reply:
So you would just rethrow AssemblyError in case of InstructionError and in case of a normal error you would just directly rethrow the normal error?

Reply:
either rethrow or make an alert and inform user that there is a bug in code where he/she has no influence on